app/src/main/java/com/asfoundation/wallet/promotions/worker/GetVipReferralWorker.kt

```diff
@@ -73,6 +80,22 @@ class GetVipReferralWorker(
       .build(),
   )
 
+  private fun getSharePendingIntent(code: String) = PendingIntent.getActivity(
+    context,
+    0,
+    ShareCompat.IntentBuilder(context)
+      .setText(code)
+      .setType("text/plain")
+      .setChooserTitle(context.getString(R.string.share_via))
+      .intent,
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
+    } else {
+      PendingIntent.FLAG_UPDATE_CURRENT
+    }
+  )
+
+
```

Comment:
Doesn't this make sense to be done in the PendingIntentNavigator like the other pending intents? And also shouldn't this be done the same way the other pendingIntents are being done? with the NavDeeplinkBuilder

Reply:
I think the NavDeepLink can't be used for this one, as it opens an android dialog to share the code, and doesn't navigate to any screen. The main one that navigates to the promotions uses the PendingIntentNavigator